### PR TITLE
Separate tests

### LINKED ISSUE #173: Decrease time to run unit tests by separating Model tests
Models like efficientdet takes a long time to test. This shouldn't be included in unittests as unittests is meant to run fast. 

Ideally pytests should separate the different kinds of tests. A suggestion is to do the following below.

`tests/unittest/pipeline/nodes/...`
`tests/integration/pipeline/nodes/...`
`tests/mltest/pipeline/models/...`

run individual tests using
`pytest tests/unittest`

Edit (28 June): more information - Another consideration is to use `pytest.mark.mltest` this [article](https://newbedev.com/how-to-keep-unit-tests-and-integrations-tests-separate-in-pytest) explains how to do this.

running with multiple marks `pytest -m "not mltest and not integration"` works. After discussions, the modularity and 'cleanliness' of this approach is preferred. Now this is the preferred approach.

### PR DESCRIPTION
**Pull request by jwaycube**
This merge separates unit tests from mlmodel tests. With this merge it will be possible to run unit test more frequently. 

closes #173

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,8 @@
 [build-system]
 requires = ["setuptools >= 40.9.0", "wheel"]
-build-backend = "setuptools.build_meta"
\ No newline at end of file
+build-backend = "setuptools.build_meta"
+
+[tool.pytest.ini_options]
+markers = [
+    "mlmodel: marks tests as (slow) ml models (deselect with '-m \"not ml\"')",
```
Comment (leeping-ng) [inline pyproject.toml:7]: Whatever's after the colon is an optional description but good to stick with the correct naming convention in the script

```suggestion
    "mlmodel: marks tests as (slow) ml models (deselect with '-m "not mlmodel"')",
```
https://docs.pytest.org/en/6.2.x/mark.html

File: tests/pipeline/nodes/model/yolov4/test_yolo.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def yolo_detector(yolo_config):
 def replace_download_weights(root, blob_file):
     return False
 
-
+@pytest.mark.unit
```
Comment (leeping-ng) [inline tests/pipeline/nodes/model/yolov4/test_yolo.py:58]: Should this be mark.mlmodel ?

File: scripts/mlmodel_tests.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+test_dir=$PWD/peekingduck
+echo "Testing code in: $test_dir"
+
+if ! (coverage run --source="$test_dir" -m pytest -m "mlmodel"); then
+    coverage report
+    echo "UNIT TESTING FAILED."
```
Comment (leeping-ng) [inline scripts/mlmodel_tests.sh:8]: Will be good for the 3 echo statements here to specify that it's testing only ml models

File: scripts/unit_tests.sh [hunk 4]
```diff
@@ -3,7 +3,7 @@
 test_dir=$PWD/peekingduck
 echo "Testing code in: $test_dir"
 
-if ! (coverage run --source=$test_dir -m pytest); then
+if ! (coverage run --source="$test_dir" -m pytest -m "not mlmodel"); then
     coverage report
     echo "UNIT TESTING FAILED."
```
Comment (leeping-ng) [inline scripts/unit_tests.sh:8]: Will be good for the 3 echo statements here to specify that it's testing everything except ml models

**Comment by leeping-ng** [comment]
Just a comment - Now there are 2 coverage reports generated from the separate unit tests - it wouldn't be accurate to take average. Perhaps when we check for coverage (which is occasional), we can remove `-m "mlmodel"` and run it manually locally?

File: pyproject.toml (hunk 1 above)
Comment (jwaycube) [reply pyproject.toml:7]: Good catch

File: tests/pipeline/nodes/model/yolov4/test_yolo.py (hunk 2 above)
Comment (jwaycube) [reply tests/pipeline/nodes/model/yolov4/test_yolo.py:58]: Yes, it was a mistake on my end. Thanks for pointing out

File: scripts/mlmodel_tests.sh (hunk 3 above)
Comment (jwaycube) [reply scripts/mlmodel_tests.sh:8]: Yes, agreed.

File: scripts/unit_tests.sh (hunk 4 above)
Comment (jwaycube) [reply scripts/unit_tests.sh:8]: Yes, agreed.

**Comment by jwaycube** [comment]
> Just a comment - Now there are 2 coverage reports generated from the separate unit tests - it wouldn't be accurate to take average. Perhaps when we check for coverage (which is occasional), we can remove `-m "mlmodel"` and run it manually locally?

Currently I am thinking that the coverage report should be run at the same frequency as the system tests. So it may be run at that level. Will need some thought for this.

Attaching issue #275 for this

**Comment by leeping-ng** [comment]
> > Just a comment - Now there are 2 coverage reports generated from the separate unit tests - it wouldn't be accurate to take average. Perhaps when we check for coverage (which is occasional), we can remove `-m "mlmodel"` and run it manually locally?
> 
> Currently I am thinking that the coverage report should be run at the same frequency as the system tests. So it may be run at that level. Will need some thought for this.
> 
> Attaching issue #275 for this

Okay, that sounds like a good approach.

**Pull request merged**